### PR TITLE
Show git trailer

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes
Create changeset trailer from description and "committed by" tag from git. Show trailers on changeset details view in a table. 

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/ui-types/src/Changesets.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import { Collection, Links } from "./hal";
 import { Tag } from "./Tags";
 import { Branch } from "./Branches";
+import { Person } from "@scm-manager/ui-components/src/avatar/Avatar";
```
Comment (sdorra) [inline scm-ui/ui-types/src/Changesets.ts:28]: We should not rely on src imports. We should export Person from `@scm-manager/ui-components` or move it to `@scm-manager/ui-types`.

File: scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+const SizedTd = styled.td`
+  width: 10rem;
+`;
+
+const ContributorTable: FC<Props> = ({ changeset }) => {
+  const [t] = useTranslation("plugins");
+
+  const collectAvailableTrailerTypes = () => {
+    // @ts-ignore
+    return [...new Set(changeset.trailers.map(trailer => trailer.trailerType))];
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx:42]: It is safe to use `Set`? Is this translated by babel?

File: scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    for (const type of availableTrailerTypes) {
+      personsByTrailerType.push({ type, persons: getPersonsByTrailersType(type) });
+    }
+    return personsByTrailerType;
+  };
+
+  return (
+    <table>
+      <tr>
+        <SizedTd>{t("changeset.trailer.type.author") + ":"}</SizedTd>
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx:62]:
```suggestion
        <SizedTd>{t("changeset.trailer.type.author")}:</SizedTd>
```

File: scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        <td>
+          <a title={changeset?.author?.mail} href={"mailto:" + changeset?.author?.mail}>
+            {changeset?.author?.name}
+          </a>
+        </td>
+      </tr>
+      {getTrailersByType().map(trailer => (
+        <tr>
+          <SizedTd>{t("changeset.trailer.type." + trailer.type) + ":"}</SizedTd>
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/components/changesets/ContributorTable.tsx:71]:
```suggestion
          <SizedTd>{t("changeset.trailer.type." + trailer.type)}:</SizedTd>
```

File: scm-webapp/src/main/java/sonia/scm/repository/ChangesetDescriptionTrailerProvider.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      foundEmptyLine = true;
+      appendLine(scanner, line);
+    }
+
+    public void appendLine(Scanner scanner, String line) {
+      newDescription.append(line);
+      if (scanner.hasNextLine()) {
+        newDescription.append('\n');
+      }
+    }
```
Comment (sdorra) [inline scm-webapp/src/main/java/sonia/scm/repository/ChangesetDescriptionTrailerProvider.java:103]: Is there are reason for make all this methods public?

Comment (pfeuffer) [reply scm-webapp/src/main/java/sonia/scm/repository/ChangesetDescriptionTrailerProvider.java]: Fixed this class.

File: scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/GitChangesetConverter.java [hunk 6]
```diff
@@ -181,6 +181,9 @@ public Changeset createChangeset(RevCommit commit, List<String> branches)
     }
 
     Changeset changeset = new Changeset(id, date, author, message);
+    if (!committerIdent.equals(authorIndent)) {
+      changeset.addTrailer(new Trailer("Committed-by", createPersonFor(committerIdent)));
```
[the remark this answers is not in the document]
Comment (pfeuffer) [reply scm-plugins/scm-git-plugin/src/main/java/sonia/scm/repository/GitChangesetConverter.java:185]: Not in the sense of git, no. Thought this is how we discussed it, though

**Comment by pfeuffer** [comment]
> Something really strange is going on:
> 
> image
> 
> Each Changeset inherits the description from the one before.

Should be fixed witch changes to `ChangesetDescriptionTrailerProvider`.

File: scm-ui/ui-webapp/public/locales/de/repos.json [hunk 7]
```diff
@@ -87,9 +87,14 @@
...
-      "prefix": "Verfasst von",
-      "mailto": "Mail senden an"
+    "contributors": {
+      "mailto": "Mail senden an",
+      "list": "Liste der Mitwirkenden",
+      "authoredBy": "Verfasst von",
+      "committedBy": "Committed von",
+      "coAuthoredBy": "Co-Autoren",
+      "more": "{{count}} mehr",
+      "count": "5 Mitwirkende"
```
Comment (pfeuffer) [inline scm-ui/ui-webapp/public/locales/de/repos.json:97]: Sure it's 5 and not 42? *scnr*

File: scm-ui/ui-webapp/public/locales/en/repos.json [hunk 8]
```diff
@@ -87,13 +87,18 @@
...
       "sources": "Sources"
+    },
+    "contributors": {
+      "mailto": "Send mail to",
+      "list": "List of contributors",
+      "authoredBy": "Authored by",
+      "committedBy": "committed by",
+      "coAuthoredBy": "co authored by",
+      "more": "{{count}} more",
+      "count": "5 Contributors"
```
Comment (pfeuffer) [inline scm-ui/ui-webapp/public/locales/en/repos.json:101]: Again, why 5?

Comment (pfeuffer) [inline scm-ui/ui-webapp/public/locales/en/repos.json:101]:
```suggestion
      "count": "{{count}} Contributors"
```

File: scm-ui/ui-webapp/public/locales/de/repos.json (hunk 7 above)
Comment (pfeuffer) [reply scm-ui/ui-webapp/public/locales/de/repos.json:97]:
```suggestion
      "count": "{{count}} Contributors"
```

**Pull request merged**